File: src/lib/email/suppression.ts
// Blocks internal / test recipients from receiving any transactional email.
// Mirrors src/trigger/_email-suppression.ts so Trigger.dev tasks and Next.js
// routes share the same rules.

const SUPPRESSED_DOMAINS = ["unclemays.com"];

const SUPPRESSED_EMAILS = new Set(
  [
    "anthonypivy@gmail.com",
    ...(process.env.EMAIL_SUPPRESSION_LIST || "")
      .split(",")
      .map((s) => s.trim().toLowerCase())
      .filter(Boolean),
  ].map((s) => s.toLowerCase())
);

export function isSuppressed(email: string | null | undefined): boolean {
  if (!email) return true;
  const normalized = email.trim().toLowerCase();
  if (!normalized) return true;
  if (SUPPRESSED_EMAILS.has(normalized)) return true;
  const domain = normalized.split("@")[1];
  return !!domain && SUPPRESSED_DOMAINS.includes(domain);
}

File: ml/ingest/_suppression.py
"""Test/internal email filter. Mirrors src/lib/email/suppression.ts so the
modeling pipeline excludes the same addresses the site excludes from
transactional sends. Anything matching here is dropped from the labeled
dataset before training.

Anthony's own test traffic (every @unclemays.com mailbox plus his personal
gmail) is the dominant noise in the historical Stripe data — without this
filter the model would learn "this looks like Anthony testing" instead of
"this is a customer who will convert."
"""

from __future__ import annotations

import os

SUPPRESSED_DOMAINS = {"unclemays.com"}

SUPPRESSED_EMAILS = {
    "anthonypivy@gmail.com",
}

# Allow extra addresses via env var (CSV).
_extra = os.environ.get("EMAIL_SUPPRESSION_LIST", "").strip()
if _extra:
    for addr in _extra.split(","):
        a = addr.strip().lower()
        if a:
            SUPPRESSED_EMAILS.add(a)


def is_suppressed(email: str | None) -> bool:
    if not email:
        return True
    norm = email.strip().lower()
    if not norm:
        return True
    if norm in SUPPRESSED_EMAILS:
        return True
    parts = norm.split("@")
    if len(parts) != 2:
        return True  # malformed
    return parts[1] in SUPPRESSED_DOMAINS
